marlowe-playground-client/src/Marlowe/ActusBlockly.purs

```diff
@@ -0,0 +1,742 @@
+    pure $ ActusAssertionNpv npv zeroInterest
+  blockDefinition _ g block = Either.Right NoActusValue
+
+instance hasBlockDefinitionPeriod :: HasBlockDefinition ActusPeriodType ActusPeriodType where
+  blockDefinition x g block = do 
+    pure $ x 
+
+
+actusDateToDay :: ActusValue -> Either String (Maybe String)
+actusDateToDay (DateValue yyyy mm dd) = Either.Right $ Just $ yyyy <> "-" <> mm <> "-" <> dd --should be validated in a parser
+actusDateToDay (ActusError msg) = Either.Left msg
+actusDateToDay NoActusValue = Either.Right Nothing
+actusDateToDay x = Either.Left $ "Unexpected: " <> show x -- should be unreachable
+
+actusDecimalToNumber :: ActusValue -> Either String (Maybe Number)
+actusDecimalToNumber (DecimalValue n) = Either.Right $ Just $ n
+actusDecimalToNumber (ActusError msg) = Either.Left msg
+actusDecimalToNumber NoActusValue = Either.Right Nothing
+actusDecimalToNumber x = Either.Left $ "Unexpected: " <> show x 
+
+blocklyCycleToCycle :: ActusValue -> Either String (Maybe Cycle)
+blocklyCycleToCycle (CycleValue _ value period) = Either.Right $ Just $ Cycle {
+  n: value 
+  , p: case period of
+     PeriodYearType -> P_Y
+     PeriodDayType -> P_D
+     PeriodMonthType -> P_M
+     PeriodQuarterType -> P_Q
+  , stub: ShortStub
+}
+blocklyCycleToCycle (ActusError msg) = Either.Left msg
+blocklyCycleToCycle NoActusValue = Either.Right Nothing
+blocklyCycleToCycle x = Either.Left $ "Unexpected: " <> show x -- should be unreachable
+
+blocklyAssertionCtxToAssertionCtx :: ActusValue -> Either String (Maybe AssertionContext)
+blocklyAssertionCtxToAssertionCtx (ActusAssertionCtx min max) = Either.Right $ Just $ AssertionContext {
+  rrmoMin: min
+  , rrmoMax: max
+}
+blocklyAssertionCtxToAssertionCtx (ActusError msg) = Either.Left msg
+blocklyAssertionCtxToAssertionCtx NoActusValue = Either.Right Nothing
+blocklyAssertionCtxToAssertionCtx x = Either.Left $ "Unexpected: " <> show x -- should be unreachable
+
+blocklyAssertionToAssertion :: ActusValue -> Either String (Maybe Assertion)
+blocklyAssertionToAssertion (ActusAssertionNpv npv rate) = Either.Right $ Just $ NpvAssertionAgainstZeroRiskBond {
+  zeroRiskInterest : rate
+  , expectedNpv : npv
+}
+blocklyAssertionToAssertion (ActusError msg) = Either.Left msg
+blocklyAssertionToAssertion NoActusValue = Either.Right Nothing
+blocklyAssertionToAssertion x = Either.Left $ "Unexpected: " <> show x -- should be unreachable
+
+blocklyCycleToAnchor :: ActusValue -> Either String (Maybe ActusValue)
+blocklyCycleToAnchor (CycleValue anchor _ _) = Either.Right $ Just anchor 
+blocklyCycleToAnchor (ActusError msg) = Either.Left msg
+blocklyCycleToAnchor NoActusValue = Either.Right Nothing
+blocklyCycleToAnchor x = Either.Left $ "Unexpected: " <> show x -- should be unreachable
+
+actusContractToTerms :: ActusContract -> Either String ContractTerms
+actusContractToTerms raw = do --todo use monad transformers?
```

Comment:
Seems like a good idea, I think this type of thing should go in a custom compiler warning if we don't want to do it in this PR but we do want to do it. `Warn (Text "We should probably use monad transformers") => ActusContract -> ...`

Reply:
I’m not sure if would look better with EitherT - I often have to distinguish between Nothing and Left. I’ll add Warn to maybe take another look at it.